localstack/services/apigateway/provider.py

```diff
@@ -89,6 +93,16 @@ def on_after_init(self):
     def _get_moto_backend(context: RequestContext) -> apigw_models.APIGatewayBackend:
         return apigw_models.apigateway_backends[context.account_id][context.region]
 
+    @staticmethod
+    def _remove_rest_api(context: RequestContext, rest_api_id: str) -> None:
+        store = get_apigateway_store(account_id=context.account_id, region=context.region)
+        # clean up this way until we properly encapsulate RestApi in the store
+        store.authorizers.pop(rest_api_id, None)
```

Comment:
Since we want to move the state from moto to localstack, in a next iteration we can functionality to the store to delete an API, basically wrapping a method to all the operation below.

Reply:
Right, I'm not sure about adding methods to the store, I've seen it done in some cases, not sure what the proper way of doing it is, any idea @thrau?